include/mbgl/map/camera.hpp

```diff
@@ -34,6 +34,31 @@ struct CameraOptions {
     /** Pitch toward the horizon measured in radians, with 0 rad resulting in a
         two-dimensional map. */
     optional<double> pitch;
+
+    bool operator==(const CameraOptions& o) const {
+        if ((!center && o.center) || (center && !o.center) ||
```

Comment:
Due to the float casts, which are needed due to slight imprecision – comparing doubles never returns true in practice. Or should a fuzzy equals be a separate method for clarity?